### PR TITLE
New documentation website

### PR DESCRIPTION
**Pull request by saghul**

### CODE REVIEW

**Comment by saghul** [comment]
@bnoordhuis PTAL! My plan here is to get the ball rolling and start filling it up (hopefully we can even get some PRs!).

The easiest way to test is with `cd docs && npm i && npm start`

File: docs/docs/developer-guide/internals.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+bytecode without a parse tree.
+
+Backtracking with an explicit stack is used so that there is no
+recursion on the system stack. Simple quantifiers are specifically
+optimized to avoid recursions.
+
+Infinite recursions coming from quantifiers with empty terms are
+avoided.
+
+The full regexp library weights about 15 KiB (x86 code), excluding the
```
Comment (bnoordhuis) [inline docs/docs/developer-guide/internals.md]:
```suggestion
The full regexp library weighs about 15 KiB (x86 code), excluding the
```

File: docs/docs/developer-guide/internals.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+A specific Unicode library was developed so that there is no
+dependency on an external large Unicode library such as ICU. All the
+Unicode tables are compressed while keeping a reasonable access
+speed.
+
+The library supports case conversion, Unicode normalization, Unicode
+script queries, Unicode general category queries and all Unicode
+binary properties.
+
+The full Unicode library weights about 45 KiB (x86 code).
```
Comment (bnoordhuis) [inline docs/docs/developer-guide/internals.md]:
```suggestion
The full Unicode library weighs about 45 KiB (x86 code).
```

File: docs/docs/es_features.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+---
+sidebar_position: 7
+---
+
+# ECMAScript Features
+
+QuickJS aims to support the latest available ECMAScript features once they hit the spec.
+
+Progress on _test262_ compliance can be checked [here](https://test262.fyi/#|qjs_ng).
```
Comment (bnoordhuis) [inline docs/docs/es_features.md:9]: It's strange, I don't see quickjs nor quickjs-ng on that site but the README in https://github.com/test262-fyi/test262.fyi does mention it and I also see references in their .github/run.yml script :shrug:

**Review by bnoordhuis — APPROVED** [review]
Nice!

File: docs/docs/es_features.md (hunk 3 above)
Comment (saghul) [reply docs/docs/es_features.md:9]: I think their last build is botched, it certainly worked a couple of days ago! I'll open an issue.

**Comment by saghul** [comment]
Thanks for taking a look! I'll follow-up with some automation to publish them.

**Comment by saghul** [comment]
✨  https://quickjs-ng.github.io/quickjs/ ✨

**Pull request merged**